src/DIRAC/Core/Utilities/JDL.py

```diff
@@ -157,3 +163,161 @@ def dumpCFGAsJDL(cfg, level=1, tab="  "):
                 contents.append("%s};" % indent)
     contents.append(f"{tab * (level - 1)}]")
     return "\n".join(contents)
+
+
+def jdlToJSON(jdlString: str):
+    """
+    Converts a JDL string into a JSON string for data validation from the BaseJob model
+    This method allows compatibility with older Client versions that used the _toJDL method
+    """
+    jdl = ClassAd(jdlString)
+
+    try:
+        jobDescription = BaseJobDescriptionModel(
+            executable=jdl.getAttributeString("Executable"),
+        )
+        if jdl.lookupAttribute("Arguments"):
+            jobDescription.arguments = str(jdl.getAttributeString("Arguments"))
+            jdl.deleteAttribute("Arguments")
+
+        if jdl.lookupAttribute("BannedSites"):
+            jobDescription.bannedSites = set(jdl.getListFromExpression("BannedSites"))
+            jdl.deleteAttribute("BannedSites")
+
+        if jdl.lookupAttribute("JobConfigArgs"):
+            jobDescription.jobConfigArgs = str(jdl.getAttributeString("JobConfigArgs"))
+            jdl.deleteAttribute("JobConfigArgs")
+
+        if jdl.lookupAttribute("CPUTime"):
+            jobDescription.cpuTime = int(jdl.getAttributeInt("CPUTime"))
+            jdl.deleteAttribute("CPUTime")
+
+        if jdl.lookupAttribute("Executable"):
+            jobDescription.executable = jdl.getAttributeString("Executable")
```

Comment:
Above you convert the result of `getAttributeString()` into a `str`. Is there any reason for not doing the same operation here?
Or the other way around: is it useful to convert it into a `str`?

```suggestion
            jobDescription.executable = str(jdl.getAttributeString("Executable"))
```

Same question for `GridCE`, `InputDataPolicy`, `JobGroup`, `JobName`, `JobType`, `LogLevel`...